### PR TITLE
image rm: allow for force-remove infra images

### PR DESCRIPTION
**Pull request by vrothberg**
Force removal of images will also remove associated containers.
Historically, infra containers have been excluded resulting in
rather annoying errors, for instance, when running `rmi -af`.

Since there is not reasons to exclude infra containers, allow for
removing the entire pod when an infra image is force removed.

Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

@containers/podman-maintainers 

I will tackle the ~same issue for `container rm` in another PR.

### CODE REVIEW

**Review by Luap99 — COMMENTED** [review]
LGTM

**Comment by vrothberg** [comment]
@rhatdan PTAL

File: test/system/010-images.bats [hunk 1]
```diff
@@ -240,4 +240,33 @@ Labels.created_at | 20[0-9-]\\\+T[0-9:]\\\+Z
...
+
+    run_podman inspect --format '{{.ID}}' $IMAGE
+    imageID=$output
+
+    run_podman version --format "{{.Server.Version}}-{{.Server.Built}}"
+    pauseImage=localhost/podman-pause:$output
+    run_podman inspect --format '{{.ID}}' $pauseImage
+    pauseID=$output
+
+    run_podman rmi -af
```
Comment (rhatdan) [inline test/system/010-images.bats:258]: Could you attempt to `run_podman rmi -a` and make sure it fails with error about the Pod, then use the force.

Comment (edsantiago) [reply test/system/010-images.bats:258]: OMG speaking of The Force, the paranoid-test-mindset is strong in this one. Great thinking, @rhatdan

**Comment by vrothberg** [comment]
This is not the test you are looking for :^) Great idea!

**Comment by mheon** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM
once the test is adjusted as suggested

**Comment by vrothberg** [comment]
Tests updated. @edsantiago @rhatdan PTAL. I figured to also exercise `rmi` without `--all` in another test.

**Comment by vrothberg** [comment]
> registry.fedoraproject.org/fedora-minimal: manifest unknown: manifest unknown

registry.fedoraproject.org is down :(

**Pull request merged**